org.eclipse.zest.core/src/org/eclipse/zest/core/viewers/internal/AbstractStructuredGraphViewer.java

```diff
@@ -250,12 +250,18 @@ GraphNode addGraphModelNode(IContainer container, Object element) {
 		}
 		return node;
 	}
+	
+	/** Implement and return the new node object, enables to define custom graph nodes*/
+	protected abstract GraphNode createNodeObject(final Graph graphModel, final int style, IFigure figure);
```

Comment:
why is this method here abstract and not as the original implementation does return the new CGraphNode. If you don't do this here clients using a different viewer wouldn't work anymore. So I think the changes you did in the GraphViewer should be done here and the GraphViewer should stay unchanged.